dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -193,9 +193,11 @@ private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody)
                             $"The request timed out, HTTP status: {response.StatusCode:G}");
 
                     case HttpStatusCode.TooManyRequests:
+                        var retryAfter = response.Headers.RetryAfter.Delta ?? response.Headers.RetryAfter.Date - DateTimeOffset.Now;
```

Comment:
could you add a comment here? it's not clear what happens 
* if the header is missing
* if the header is malformed
* if the header value is wrong, e.g. "retry in 1 month" or "retry in -1 minute"

Reply:
I can look into the docs more, not sure what it does, but I think these properties are worth using.

https://learn.microsoft.com/en-us/dotnet/api/system.net.http.headers.retryconditionheadervalue?view=net-7.0

Reply:
And add some comments.